bolt-sidecar/src/crypto/bls.rs

```diff
@@ -49,6 +50,25 @@ pub trait SignerBLSAsync: Send + Sync {
     async fn sign(&self, data: &[u8]) -> eyre::Result<BLSSig>;
 }
 
+/// Enum to represent the two possible signer types.
+#[derive(Debug)]
+pub enum BlsSignerType {
+    /// A BLS signer that uses private key to sign.
+    PrivateKey(Signer),
+    /// A BLS signer that request signature from commit-boost.
+    CommitBoost(CommitBoostClient),
+}
```

Comment:
Not fully sure if this is the best way to support both, I tried doing some other ways but this seemed to me the most cleaner